src/components/__tests__/RouteItemDisplay.cy.js

```diff
@@ -142,20 +156,124 @@ function coreTests() {
     cy.dataCy('icon-transport')
       .should('be.visible')
-      .invoke('height')
-      .should('be.eq', 24);
-    cy.dataCy('icon-transport')
-      .should('be.visible')
-      .invoke('width')
-      .should('be.eq', 24);
+      .and('have.css', 'font-size', '18px');
+    cy.dataCy('icon-transport').invoke('height').should('be.eq', 18);
+    cy.dataCy('icon-transport').invoke('width').should('be.eq', 18);
     // description transport styles
     cy.dataCy('description-transport')
       .should('be.visible')
       .and('have.css', 'font-size', '14px')
       .and('have.css', 'font-weight', '400')
-      .and('have.color', black);
+      .and('have.color', grey10);
+  });
+
+  it('renders correct transport value', () => {
+    cy.get('@route').then((route) => {
+      // test styles if transport value is logged
+      if (route.transport) {
+        // transport value
+        cy.dataCy('avatar-transport')
+          .should('be.visible')
+          .and('have.backgroundColor', secondary);
+        // icon transport
+        cy.dataCy('icon-transport')
+          .should('be.visible')
+          .and('have.color', primary);
+      } else {
+        // transport value empty
+        cy.dataCy('avatar-transport')
+          .should('be.visible')
+          .and('have.backgroundColor', grey2);
+        // icon transport question mark
+        cy.dataCy('icon-transport')
+          .should('be.visible')
+          .and('have.color', grey7);
+      }
+      cy.dataCy('description-transport').then((description) => {
+        cy.wrap(getTransportLabel(route.transport)).should(
+          'eq',
+          description.text(),
+        );
+      });
+    });
+  });
+
+  it('renders two sections stacked', () => {
+    cy.testElementPercentageWidth(cy.dataCy('section-direction'), 100);
+    cy.testElementPercentageWidth(cy.dataCy('section-distance'), 100);
+  });
+
+  it('renders correct distance value', () => {
+    cy.get('@route').then((route) => {
+      if (route.distance) {
+        // distance value including units
+        cy.dataCy('label-distance')
+          .should('be.visible')
+          .and('have.css', 'font-size', '14px')
+          .and('have.css', 'font-weight', '700')
+          .and('have.color', grey10)
+          .and('contain', route.distance)
+          .and('contain', i18n.global.t('global.routeLengthUnit'));
+      } else {
+        // distance value empty
+        cy.dataCy('label-distance').should('not.exist');
+      }
+    });
+  });
+
+  it('renders icons correctly', () => {
+    // icon direction
+    cy.dataCy('label-direction-icon', { timeout: 4000 })
+      .should('be.visible')
+      .matchImageSnapshot(`${Cypress.currentTest.titlePath}-direction`, {
+        failureThreshold: 0.1,
+        failureThresholdType: 'percent',
+        timeout: 4000,
+        customDiffConfig: { threshold: 0.4 },
+        retries: 2,
+      });
+    // icon transport
+    cy.dataCy('icon-transport', { timeout: 4000 })
```

Comment:
Really we need timeout here?

Reply:
This was more of a precaution. Local tests pass without it. Removed.